scene/gui/control.cpp

```diff
@@ -742,11 +738,14 @@ void Control::set_anchor(Side p_side, real_t p_anchor, bool p_keep_offset, bool
 	}
 
 	if (!p_keep_offset) {
-		data.offset[p_side] = previous_pos - data.anchor[p_side] * parent_range;
-		if (p_push_opposite_anchor) {
-			data.offset[(p_side + 2) % 4] = previous_opposite_pos - data.anchor[(p_side + 2) % 4] * parent_range;
-		}
+		if(p_side == SIDE_LEFT) {data.offset[0] = 0;};
+		if(p_side == SIDE_TOP) {data.offset[1] = 0;};
+		if(p_side == SIDE_RIGHT) {data.offset[2] = (data.anchor[2] - 1) * (parent_rect.size.x - parent_rect.position.x);};
+		if(p_side == SIDE_BOTTOM) {data.offset[3] = (data.anchor[3] - 1) * (parent_rect.size.y - parent_rect.position.y);};
 	}
+
+	
+
```

Comment:
```suggestion
		if (p_side == SIDE_LEFT) {
			data.offset[0] = 0;
		}
		if (p_side == SIDE_TOP) {
			data.offset[1] = 0;
		}
		if (p_side == SIDE_RIGHT) {
			data.offset[2] = (data.anchor[2] - 1) * (parent_rect.size.x - parent_rect.position.x);
		}
		if (p_side == SIDE_BOTTOM) {
			data.offset[3] = (data.anchor[3] - 1) * (parent_rect.size.y - parent_rect.position.y);
		}
	}

```
Code style

Reply:
Yeah! Sorry, first time doing a PR and using cpp